tests/utilities/fault_tolerant.py

```diff
@@ -0,0 +1,142 @@
+# Copyright The PyTorch Lightning team.
```

Comment:
Our CI will not test this, right? Why should we have it here then?

Reply:
This is used by Grid to validate its fault tolerant CI works as expected with Lightning.
And this script should be used by any other platform to provide a seamless experience with Fault Tolerant Training and Lightning.

Reply:
But how does Grid run it? Does it clone the repository just to access this file?

Why can't it be a `gist` then? It would be more convenient to update then.

Reply:
Grid clone Pytorch Lightning. I am still debuting it right now.

Reply:
If our CI does not run it, I would not add it here.

Normally, it would be in the Grid codebase, but if you expect other platforms to use it, then I'd publish it as a public gist.

This has the advantage of not having to get reviews + 3 approvals in Lightning for any change. If Grid CI breaks because this fails and the fix is in this script in particular, you want to be able to move fast.

Reply:
Improve the code and moved it as an example.

Reply:
Done !